src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -1046,17 +1049,47 @@ func (c *controller) updateNodesLabels() bool {
 			continue
 		}
 		hostsWithLabels[hostname] = hostData
+		allNodeLabelsAsOneString += hostData.Host.NodeLabels
+	}
+	return hostsWithLabels, strings.Contains(allNodeLabelsAsOneString, roleLabel), nil
+}
+
+// Set labels to hosts
+// If one of the labels was role pause mcp and set this role
+// after setting all labels unpause in case it was paused
+func (c *controller) setNodesLabels() bool {
+	c.log.Infof("Setting node labels if require")
+	hostsWithLabels, pauseMCO, err := c.getAssistedHostsWithLabels()
```

Comment:
The decision if to pause/unpause should mostly rely on the machine_config_pool_name.  If not then there is no reason to do it.

Reply:
It is difficult decision as pool can have different names from role as far as i understand and just have relevant labels. I actually don't think we should go and list all the pools in order to check if one of them have relevant labels inside. 
Not sure it worth such complication

Reply:
If exists(machine_config_pool_name) != "" ?

Reply:
Checking machine_config_pool_name is an indication that the user actually wanted to use different MCP.  Otherwise it can be just a label.

Reply:
But do it requires same pool name as role? if not it is not going to be that easy. Though we can check that there is more than 2 mcps. To say the truth i don't see any problem in pausing mco as it does nothing if we unpause it at the end

Reply:
I think we should not complicate it now and see how it is going.